junit_tests/DateTimeTest.java

```diff
@@ -0,0 +1,707 @@
+        Str string = new Str("string");
+        Exception exception = assertThrows(TypeError.class, () -> {
+            dateTime.__lt__(string);
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+
+        expectedMessage = "'<' not supported between instances of 'datetime.datetime' and 'int'";
+        Int integer = Int.getInt(1);
+        exception = assertThrows(TypeError.class, () -> {
+            dateTime.__lt__(integer);
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+
+        expectedMessage = "'<' not supported between instances of 'datetime.datetime' and 'float'";
+        Float fl = new Float(1.0);
+        exception = assertThrows(TypeError.class, () -> {
+            dateTime.__lt__(fl);
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    public void testLeCombinations() {
+        List<DateTime> sortedDateTimes = getSortedDateTimes();
+        for (int i = 0; i < 7; i++) {
+            for (int j = 0; j < 7; j++) {
+                if (i <= j) {
+                    assertEquals(Bool.getBool(true), sortedDateTimes.get(i).__le__(sortedDateTimes.get(j)));
+                } else {
+                    assertEquals(Bool.getBool(false), sortedDateTimes.get(i).__le__(sortedDateTimes.get(j)));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testLeUnsupportedTypes() {
+        String expectedMessage = "'<=' not supported between instances of 'datetime.datetime' and 'str'";
+        Str string = new Str("string");
+        Exception exception = assertThrows(TypeError.class, () -> {
+            dateTime.__le__(string);
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+
+        expectedMessage = "'<=' not supported between instances of 'datetime.datetime' and 'int'";
+        Int integer = Int.getInt(1);
+        exception = assertThrows(TypeError.class, () -> {
+            dateTime.__le__(integer);
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+
+        expectedMessage = "'<=' not supported between instances of 'datetime.datetime' and 'float'";
+        Float fl = new Float(1.0);
+        exception = assertThrows(TypeError.class, () -> {
+            dateTime.__le__(fl);
+        });
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    public void testEqCombinations() {
+        List<DateTime> sortedDateTimes = getSortedDateTimes();
+        for (int i = 0; i < 7; i++) {
+            for (int j = 0; j < 7; j++) {
+                if (i == j) {
+                    assertEquals(Bool.getBool(true), sortedDateTimes.get(i).__eq__(sortedDateTimes.get(j)));
+                } else {
+                    assertEquals(Bool.getBool(false), sortedDateTimes.get(i).__eq__(sortedDateTimes.get(j)));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testEqUnsupportedTypes() {
```

Comment:
You test using strings each time, you should use int and float for the second and third assert

Reply:
Good catch. Fixed in 946bb34.